com.unity.multiplayer.mlapi/Runtime/Core/NetworkObject.cs

```diff
@@ -333,7 +333,7 @@ public static void NetworkHide(List<NetworkObject> networkObjects, ulong clientI
 
         private void OnDestroy()
         {
-            if (NetworkManager.Singleton != null && NetworkManager.Singleton.SpawnManager.SpawnedObjects.ContainsKey(NetworkObjectId))
+            if (NetworkManager.Singleton != null && NetworkManager.Singleton.SpawnManager != null && NetworkManager.Singleton.SpawnManager.SpawnedObjects.ContainsKey(NetworkObjectId))
```

Comment:
if this fix does not prevent this feature from shipping, it should simply go under a different PR.
(so that we can discuss more in details over there)